lib/ecto/migrator.ex

```diff
@@ -542,18 +546,19 @@ defmodule Ecto.Migrator do
     |> Enum.reverse
   end
 
-  # This function will match directories passed into `Migrator.run`.
-  defp migrations_for(migration_source) when is_binary(migration_source) do
-    Path.join([migration_source, "**", "*.exs"])
-    |> Path.wildcard()
-    |> Enum.map(&extract_migration_info/1)
-    |> Enum.filter(& &1)
-    |> Enum.sort()
-  end
-
-  # This function will match specific version/modules passed into `Migrator.run`.
   defp migrations_for(migration_source) when is_list(migration_source) do
-    Enum.map migration_source, fn {version, module} -> {version, module, module} end
+    Enum.map(migration_source, fn
+      directory when is_binary(directory) ->
+        Path.join([directory, "**", "*.exs"])
+        |> Path.wildcard()
+        |> Enum.map(&extract_migration_info/1)
+        |> Enum.filter(& &1)
+
+      {version, module} ->
+        {version, module, module}
+    end)
+    |> List.flatten()
```

Comment:
Use `flat_map` instead.

Reply:
Did you mean to literally replace `|> List.flatten()` with `|> Enum.flat_map(& &1)` or using some of the logic as the argument to the flat_map?

Reply:
I guess you meant this?

```elixir
    migration_source
    |> Enum.flat_map(fn
      directory when is_binary(directory) ->
        Path.join([directory, "**", "*.exs"])
        |> Path.wildcard()
        |> Enum.map(&extract_migration_info/1)
        |> Enum.filter(& &1)

      {version, module} ->
        [{version, module, module}]
    end)
    |> Enum.sort()
```

Reply:
Correct.